modules/fbx/data/fbx_mesh_data.cpp

```diff
@@ -1092,7 +1092,10 @@ HashMap<int, R> FBXMeshData::extract_per_vertex_data(
 					const int vertex_index = get_vertex_from_polygon_vertex(p_mesh_indices, polygon_vertex_index);
 					ERR_FAIL_COND_V_MSG(vertex_index < 0, (HashMap<int, R>()), "FBX file corrupted: #ERR05");
 					ERR_FAIL_COND_V_MSG(vertex_index >= p_vertex_count, (HashMap<int, R>()), "FBX file corrupted: #ERR06");
-					const int index_to_direct = p_mapping_data.index[polygon_vertex_index];
+					int index_to_direct = p_mapping_data.index[polygon_vertex_index];
+					if (index_to_direct < 0) {
+						index_to_direct = ~index_to_direct; // Relative to the end.
```

Comment:
CC @RevoluPowered 

Are you sure that this always give a valid index and the one expected?
Other places in this file seem to have ERR conditions for negative `p_mapping_data.index[vertex_index]`.

Reply:
Negative indices are only allowed when handling polygons according to [this tutorial](https://banexdevblog.wordpress.com/2014/06/23/a-quick-tutorial-about-the-fbx-ascii-format/) and the [blender fbx importer](https://github.com/blender/blender-addons/blob/40e34a8bbb30c6a4ddc73aa11e92c6bd2095eb80/io_scene_fbx/import_fbx.py#L1269-L1280).

These touched codes are for `MapType::polygon_vertex` and `MapType::polygon`.

This one is for `MapType::vertex`:

https://github.com/godotengine/godot/blob/d7464757084099765dee5755b21a8847f4226360/modules/fbx/data/fbx_mesh_data.cpp#L1078

Reply:
Ah, I think `FBXMeshData::get_vertex_from_polygon_vertex()` is an existing wrapper code for this logic. I should change the code to use that directly.